render/wgpu/src/backend.rs

```diff
@@ -917,7 +919,66 @@ impl<T: RenderTarget + 'static> RenderBackend for WgpuRenderBackend<T> {
         arguments: &[PixelBenderShaderArgument],
         target_handle: BitmapHandle,
     ) -> Result<Box<dyn SyncHandle>, BitmapError> {
-        self.run_pixelbender_shader_impl(shader, arguments, target_handle)
+        let target = as_texture(&target_handle);
+
+        let extent = wgpu::Extent3d {
+            width: target.texture.width(),
+            height: target.texture.height(),
+            depth_or_array_layers: 1,
+        };
+
+        let buffer_info = self.get_texture_buffer_info(
+            target,
+            PixelRegion::for_whole_size(target.texture.width(), target.texture.height()),
+        );
+
+        let mut texture_target = TextureTarget {
+            size: extent,
+            texture: target.texture.clone(),
+            format: wgpu::TextureFormat::Rgba8Unorm,
+            buffer: buffer_info,
+        };
+
+        let frame_output = texture_target
+            .get_next_texture()
+            .expect("TextureTargetFrame.get_next_texture is infallible");
+
+        let mut render_command_encoder =
+            self.descriptors
+                .device
+                .create_command_encoder(&wgpu::CommandEncoderDescriptor {
+                    label: create_debug_label!("Render command encoder").as_deref(),
+                });
+
+        run_pixelbender_shader_impl(
+            &self.descriptors,
+            shader,
+            ShaderMode::ShaderJob,
+            arguments,
+            &target.texture,
+            &mut render_command_encoder,
+            Some(wgpu::RenderPassColorAttachment {
+                view: frame_output.view(),
+                resolve_target: None,
+                ops: wgpu::Operations {
+                    load: wgpu::LoadOp::Clear(wgpu::Color::TRANSPARENT),
+                    store: true,
+                },
+            }),
+            // When running a standalone shader, we always process the entire image
+            &FilterSource::for_entire_texture(&target.texture),
+        )?;
+
+        self.descriptors
+            .queue
+            .submit(Some(render_command_encoder.finish()));
+
+        Ok(Box::new(QueueSyncHandle::NotCopied {
```

Comment:
This now needs to be `QueueSyncHandle::AlreadyCopied` depending on target buffer